lib/internal/fs/promises.js

```diff
@@ -687,7 +689,11 @@ async function readlink(path, options) {
 }
 
 async function symlink(target, path, type_) {
-  const type = (typeof type_ === 'string' ? type_ : null);
+  let type = (typeof type_ === 'string' ? type_ : null);
```

Comment:
Since this is semver-major anyway, we might as well use `validateString` here, wdyt?

Reply:
Agreed. Something like `validateOneOf(type, 'type', ['dir', 'file', 'junction', undefined])` would be perfect.
I'd also prefer to have this validation performed on all platforms for consistency.
However, technically it can break userland code (e.g. if someone used `null` or `false`), so full deprecation cycle might be safer.